Source/DafnyLanguageServer/Handlers/DafnyCodeActionHandler.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System.Collections.Concurrent;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.Plugins;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Dafny.LanguageServer.Handlers; 
+
+public class DafnyCodeActionHandler : CodeActionHandlerBase {
+  private readonly ILogger<DafnyCodeActionHandler> logger;
+  private readonly IDocumentDatabase documents;
+
+  public DafnyCodeActionHandler(ILogger<DafnyCodeActionHandler> logger, IDocumentDatabase documents) {
+    this.logger = logger;
+    this.documents = documents;
+  }
+
+  public record QuickFixWithId(QuickFix QuickFix, int Id);
+
+  protected override CodeActionRegistrationOptions CreateRegistrationOptions(CodeActionCapability capability,
+    ClientCapabilities clientCapabilities) {
+    return new CodeActionRegistrationOptions {
+      DocumentSelector = DocumentSelector.ForLanguage("dafny"),
+      ResolveProvider = true,
+      CodeActionKinds = Container<CodeActionKind>.From(
+        CodeActionKind.QuickFix
+      ),
+      WorkDoneProgress = false
+    };
+  }
+
+
+  /// <summary>
+  /// Returns the fixes along with a unique identifier
+  /// </summary>
+  public IEnumerable<QuickFixWithId> GetFixesWithIds(QuickFixer[] fixers, DafnyDocument document, CodeActionParams request) {
```

Comment:
I think the input argument `fixers` here should be of type `IEnumerable<QuickFixer>` to be more admissible.

Reply:
> I think the input argument fixers here should be of type IEnumerable<QuickFixer> to be more admissible.

I also made the method private because it does not need to be public.